File: LaoCai_2324/Bai5/bai5.cpp
#include <iostream>
#include <vector>
#include <algorithm>
#include <fstream>

using namespace std;

int main() {
    // Đọc dữ liệu từ tệp bai5.inp
    ifstream infile("bai5.inp");
    ofstream outfile("bai5.out");

    int n;
    infile >> n; // Đọc n
    vector<int> A(n);

    // Đọc dãy số A
    for (int i = 0; i < n; i++) {
        infile >> A[i];
    }

    // Sắp xếp dãy số A
    sort(A.begin(), A.end());

    int max_mod = 0; // Biến lưu giá trị lớn nhất của ai mod aj

    // Duyệt qua từng phần tử trong dãy đã sắp xếp
    for (int i = 1; i < n; i++) { // Bắt đầu từ 1 vì cần có aj
        for (int j = 0; j < i; j++) {
            // Tính ai mod aj
            int mod_value = A[i] % A[j];
            // Cập nhật giá trị lớn nhất
            if (mod_value > max_mod) {
                max_mod = mod_value;
            }
        }
    }

    // Ghi kết quả vào tệp bai5.out
    outfile << max_mod << endl;

    infile.close();
    outfile.close();

    return 0;
}

File: THCS/2023_2024/LaoCai_2324/Bai5/bai5.py
def main():
    # Đọc dữ liệu từ tệp bai5.inp
    with open("bai5.inp", "r") as infile:
        n = int(infile.readline().strip())  # Đọc n
        A = list(map(int, infile.readline().strip().split()))  # Đọc dãy số A

    # Sắp xếp dãy số A
    A.sort()

    max_mod = 0  # Biến lưu giá trị lớn nhất của ai mod aj

    # Duyệt qua từng phần tử trong dãy đã sắp xếp
    for i in range(n):
        for j in range(i):
            # Tính ai mod aj
            mod_value = A[i] % A[j]
            # Cập nhật giá trị lớn nhất
            if mod_value > max_mod:
                max_mod = mod_value

    # Ghi kết quả vào tệp bai5.out
    with open("bai5.out", "w") as outfile:
        outfile.write(str(max_mod) + "\n")

if __name__ == "__main__":
    main()
